src/server.rs

```diff
@@ -119,4 +123,20 @@ impl Hopper {
             tokio::task::yield_now().await
         }
     }
+
+    pub(crate) fn listen_config(&self, mut stream: Signal) {
+        let router = self.router.clone();
+        tokio::spawn(async move {
+            loop {
+                stream.recv().await;
+
+                let config = ServerConfig::read().expect("Unable to read config");
```

Comment:
When `expect` panics the task stops but the process lives on, thus making hopper incapable of reloading new configurations.